zanata-war/src/test/java/org/zanata/service/impl/MergeTranslationsServiceImplTest.java

```diff
@@ -0,0 +1,307 @@
+    }
+
+    // target tft is has same modify date as source tft
+    @Test
+    public void testMergeTranslationWorkCheckDate1() {
+        Date now = new Date();
+        HTextFlowTarget target1 = generateTarget(ContentState.Translated, now, "string1");
+        HTextFlowTarget target2 = generateTarget(ContentState.Translated, now, "string2");
+
+        testShouldMergeCondition(target1, target2, false, false);
+    }
+
+    // target tft is newer than source tft
+    @Test
+    public void testMergeTranslationWorkCheckDate2() {
+        Calendar c = new GregorianCalendar();
+        HTextFlowTarget target1 =
+                generateTarget(ContentState.Translated, c.getTime(), "string1");
+        c.add(Calendar.DATE, 30);
+        HTextFlowTarget target2 =
+                generateTarget(ContentState.Translated, c.getTime(), "string2");
+
+        testShouldMergeCondition(target1, target2, true, false);
+    }
+
+    // target tft is older than source tft
+    @Test
+    public void testMergeTranslationWorkCheckDate3() {
+        Calendar c = new GregorianCalendar();
+        HTextFlowTarget target2 =
+                generateTarget(ContentState.Translated, c.getTime(), "string1");
+        c.add(Calendar.DATE, 30);
+        HTextFlowTarget target1 =
+                generateTarget(ContentState.Translated, c.getTime(), "string2");
+
+        testShouldMergeCondition(target1, target2, true, true);
+    }
+
+    // target tft is older than source tft, but same content
+    @Test
+    public void testMergeTranslationWorkSameStateAndContent() {
+        String content = "content0";
+        Calendar c = new GregorianCalendar();
+        HTextFlowTarget target2 =
+            generateTarget(ContentState.Translated, c.getTime(), content);
+        c.add(Calendar.DATE, 30);
+        HTextFlowTarget target1 =
+            generateTarget(ContentState.Translated, c.getTime(), content);
+        testShouldMergeCondition(target1, target2, false, false);
+    }
+
+    private HTextFlowTarget generateTarget(ContentState state,
+            Date lastChanged, String content) {
+        HTextFlowTarget target = new HTextFlowTarget();
+        target.setState(state);
+        target.setLastChanged(lastChanged);
+        target.setContents(content);
+        return target;
+    }
+
+    private void testShouldMergeCondition(HTextFlowTarget target1,
+            HTextFlowTarget target2, boolean useNewerTranslation,
+            boolean expectedResult) {
+        boolean result =
+                service.shouldMerge(target1, target2, useNewerTranslation);
+        assertThat(result).isEqualTo(expectedResult);
+    }
+
+    @Test
+    public void testDocIdAndContentHash() {
+        /**
+         * this test is covered by @see org.zanata.dao.TextFlowTargetDAOTest#
+         * testGetTranslationsByMatchedContext
+         *
+         * check different docId won't copy check different tf.contentHash won't
+         * copy
+         */
+    }
```

Comment:
I think an empty test is not the way to communicate this. Probably better just to make this a comment, maybe in the doc comment in this class.